lib/Backend/arm64/Thunks.asm

```diff
@@ -26,40 +26,36 @@
     ;Js::Var NativeCodeGenerator::CheckCodeGenThunk(Js::RecyclableObject* function, Js::CallInfo callInfo, ...)
     NESTED_ENTRY ?CheckCodeGenThunk@NativeCodeGenerator@@SAPEAXPEAVRecyclableObject@Js@@UCallInfo@3@ZZ
 
-    PROLOG_SAVE_REG_PAIR fp, lr, #-(16*8+32)!          ; establish stack frame
-    PROLOG_SAVE_REG_PAIR x19, x20, #16
-    stp     x0, x1, [sp, #32+0*8]
-    stp     x2, x3, [sp, #32+2*8]
-    stp     x4, x5, [sp, #32+4*8]
-    stp     x6, x7, [sp, #32+6*8]
-    stp     x8, x9, [sp, #32+8*8]
-    stp     x10, x11, [sp, #32+10*8]
-    stp     x12, x13, [sp, #32+12*8]
-    stp     x14, x15, [sp, #32+14*8]
+    PROLOG_SAVE_REG_PAIR fp, lr, #-(2*8+16*8)!          ; establish stack frame
+    stp     d0, d1, [sp, #16+0*8]
+    stp     d2, d3, [sp, #16+2*8]
+    stp     d4, d5, [sp, #16+4*8]
+    stp     d6, d7, [sp, #16+6*8]
```

Comment:
In other thunks in arm64_thunks.asm, we only save and restore x0-x7 (apart from fp, lr). Should those thunks also save and restore d0-d7?

Reply:
It depends on the function you are calling through to. If it's a varargs target, then we only need to worry about x0-x7. If there's a possibility we could be calling a function that takes an float/double parameters, then we definitely need to preserve d0-d7. I might be being a bit too paranoid here.